### DISCUSSION TITLE
如何配置？

### DISCUSSION THREAD

**Discussion by Lynricsy** [spencerwooo/onedrive-vercel-index#9] [Q&A] [answered]
没找到配置client_secret和refresh_token的地方啊

**Comment by Revincx** [comment]
client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
cilent_secret和refresh_token在环境变量里。

**Comment by Lynricsy** [comment]
> client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
> cilent_secret和refresh_token在环境变量里。

好的，谢谢！

**Comment by Lynricsy** [comment]
> client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
> cilent_secret和refresh_token在环境变量里。

环境变量名字是什么呢？

**Comment by Revincx** [comment]
> > client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
> > cilent_secret和refresh_token在环境变量里。
> 
> 环境变量名字是什么呢？

你看下[这一行](https://github.com/spencerwooo/onedrive-vercel-index/blob/c1cdecae7ff88492c7612ff6199880fbd0849f34/pages/api/index.ts#L25)就知道了

**Comment by Lynricsy** [comment]
> > > client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
> > > cilent_secret和refresh_token在环境变量里。
> > 
> > 
> > 环境变量名字是什么呢？
> 
> 你看下[这一行](https://github.com/spencerwooo/onedrive-vercel-index/blob/c1cdecae7ff88492c7612ff6199880fbd0849f34/pages/api/index.ts#L25)就知道了

嗯嗯……自己懒得看代码，所以就来问了，谢谢

**Comment by Lynricsy** [comment]
> > > client_id在[/config/api.json](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/config/api.json)里
> > > cilent_secret和refresh_token在环境变量里。
> > 
> > 
> > 环境变量名字是什么呢？
> 
> 你看下[这一行](https://github.com/spencerwooo/onedrive-vercel-index/blob/c1cdecae7ff88492c7612ff6199880fbd0849f34/pages/api/index.ts#L25)就知道了

可是配好了仍然Error: Request failed with status code 500

**Comment by spencerwooo** [comment, marked as answer]
部署方法简单写在 [README 的 deployment 部分](https://github.com/spencerwooo/onedrive-vercel-index#deployment)，如果你遇到 API 问题那么还是你自己的 token 权限配置的 bug。

**Comment by baker221** [comment]
请问 `ACCESS_TOKEN` 是从哪里获得的呢？我在源项目[onedrive-cf-index](https://github.com/spencerwooo/onedrive-cf-index/blob/main/README-CN.md) 的 README 中并未看到该变量从哪里获得

**Reply by aozaki-kuro** [reply]
看 [这一节](https://github.com/spencerwooo/onedrive-cf-index/blob/main/README-CN.md#%E7%94%9F%E6%88%90-onedrive-api-%E4%BB%A4%E7%89%8C) 的第五点
```bash
npx @beetcb/ms-graph-cli
```
是用这个部署的。

**Comment by Tokisaki-Galaxy** [comment]
环境变量写不下了怎么办？
Error:The total size of all Environment Variables (4.09KB) exceeds 4KB
